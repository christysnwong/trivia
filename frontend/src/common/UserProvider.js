import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
