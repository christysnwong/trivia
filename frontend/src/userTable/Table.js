import TableHeading from "./TableHeading";
import TableRow from "./TableRow";


const Table = ({entries, leaderboard}) => {

  return (
    <div className="table-responsive">
      <table className="table table-striped table-sm">
        <thead>
          <tr>
            <th scope="col">#</th>
            {!!entries.length &&
              Object.keys(entries[0]).map((heading, idx) => (
                <TableHeading key={idx} heading={heading} />
              ))}
          </tr>
        </thead>
        <tbody>
          {entries.map((entry, idx) => (
            <TableRow
              key={idx}
              id={idx}
              entry={entry}
              leaderboard={leaderboard}
            />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Table;


