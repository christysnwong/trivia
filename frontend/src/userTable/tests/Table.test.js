import React from "react";
import { render } from "@testing-library/react";
import Table from "../Table";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../../common/UserProvider";

const entries = [
  {
    category: "Entertainment: Books",
    difficulty: "easy",
    score: 9,
    points: 70,
    date: "Sep 8th 2022, 1:20pm",
  },
  {
    category: "Entertainment: Film",
    difficulty: "easy",
    score: 9,
    points: 70,
    date: "Sep 8th 2022, 12:20pm",
  },
  {
    category: "Entertainment: Film",
    difficulty: "medium",
    score: 10,
    points: 70,
    date: "Sep 8th 2022, 1:06pm",
  },
];

it("renders without crashing", function () {
  render(
    <UserProvider>
      <Table entries={entries} />
    </UserProvider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Table entries={entries} />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
