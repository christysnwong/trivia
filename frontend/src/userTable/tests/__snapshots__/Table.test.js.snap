// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table table-striped table-sm"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            #
          </th>
          <th
            class="text-capitalize fw-bold"
            scope="col"
          >
            category
          </th>
          <th
            class="text-capitalize fw-bold"
            scope="col"
          >
            difficulty
          </th>
          <th
            class="text-capitalize fw-bold"
            scope="col"
          >
            score
          </th>
          <th
            class="text-capitalize fw-bold"
            scope="col"
          >
            points
          </th>
          <th
            class="text-capitalize fw-bold"
            scope="col"
          >
            date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            scope="row"
          >
            1
          </th>
          <td
            class="text-capitalize"
          >
            Entertainment: Books
          </td>
          <td
            class="text-capitalize"
          >
            easy
          </td>
          <td
            class="text-capitalize"
          >
            9
          </td>
          <td
            class="text-capitalize"
          >
            70
          </td>
          <td
            class="text-capitalize"
          >
            Sep 8th 2022, 1:20pm
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            2
          </th>
          <td
            class="text-capitalize"
          >
            Entertainment: Film
          </td>
          <td
            class="text-capitalize"
          >
            easy
          </td>
          <td
            class="text-capitalize"
          >
            9
          </td>
          <td
            class="text-capitalize"
          >
            70
          </td>
          <td
            class="text-capitalize"
          >
            Sep 8th 2022, 12:20pm
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            3
          </th>
          <td
            class="text-capitalize"
          >
            Entertainment: Film
          </td>
          <td
            class="text-capitalize"
          >
            medium
          </td>
          <td
            class="text-capitalize"
          >
            10
          </td>
          <td
            class="text-capitalize"
          >
            70
          </td>
          <td
            class="text-capitalize"
          >
            Sep 8th 2022, 1:06pm
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
